tests/bson/bson-utcdatetimeinterface-002.phpt

```diff
@@ -0,0 +1,105 @@
+--TEST--
+MongoDB\BSON\UTCDateTimeInterface example with MongoDB\BSON\TypeWrapper
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+class MyUTCDateTime implements MongoDB\BSON\TypeWrapper, MongoDB\BSON\UTCDateTimeInterface
+{
+    private $date;
+
+    public function __construct($milliseconds)
+    {
+        $this->date = new MongoDB\BSON\UTCDateTime($milliseconds);
+    }
+
+    public static function createFromBSONType(MongoDB\BSON\Type $type)
+    {
+        if ( ! $type instanceof MongoDB\BSON\UTCDateTime) {
+            throw new InvalidArgumentException('Cannot create MyUTCDateTime from ' . get_class($type));
+        }
+
+        return new self((string) $type);
```

Comment:
A string cast is necessary to avoid `UTCDateTime::__construct()` throwing an exception because its object argument is not a DateTime instance. This issue didn't come up in the Decimal128 and ObjectID tests, as we only parse the argument as a string and PHP happily calls `__toString()` on the object argument for us.

I don't think this is anything we need to fix, as this is all consistent with UTCDateTime's documentation and the Decimal128 and ObjectID behavior is just a happy coincidence.